src/transitions.rs

```diff
@@ -273,23 +273,14 @@ impl FixedTimespanSetBuilder {
                     self.until_time = Some(timespan.end_time.unwrap().to_timestamp() - utc_offset - *dst_offset);
                 }
 
-                // Find the minimum rule based on the current UTC and DST offsets.
-                // (this can be replaced with min_by when it stabilises):
-                //.min_by(|r| r.1.absolute_datetime(year, utc_offset, dst_offset));
-                let pos = {
-                    let earliest = activated_rules.iter().enumerate()
-                        .map(|(i, r)| (r.absolute_datetime(year, utc_offset, *dst_offset), i))
-                        .min()
-                        .map(|(_, i)| i);
-
-                    match earliest {
-                        Some(p) => p,
-                        None    => break,
-                    }
-                };
-
-                let earliest_rule = activated_rules.remove(pos);
-                let earliest_at = earliest_rule.absolute_datetime(year, utc_offset, *dst_offset);
+                // Find the minimum rule and its start time based on the current UTC and DST offsets.
+                let (i, earliest_at) = match activated_rules.iter().enumerate()
```

Comment:
Could you assign the result of the `iter...min_by_key` to a variable to make this match a bit more obvious?

Could you also leave `i` as `pos` here, to reduce the churn a bit, and to make it more clear that we're looking for a position?

Reply:
I left the min element from the iter named the same `earliest` (I think this is the one that flows best) and applied the other suggestions. Let me know if this is more readable.